examples/io_test/io_read.f90

```diff
@@ -91,8 +91,8 @@ program io_read
 
 #ifndef ADIOS2
    if (nrank == 0) then
-      inquire (file="out", exist=dir_exists)
-      if (.not. dir_exists) then
+      inquire (file="out/u1.dat", exist=file_exists)
```

Comment:
Ah, I hadn't realised this was the test the issue was in. I think we should check existence of all the files to be read in.